### PR TITLE
Update to gradle 7.6 + other deps + just a lot of general yak shaving

### LINKED ISSUE #96: Use new ArtifactCollection APIs in a future gradle version
https://github.com/gradle/gradle/pull/18729 should unlock some things for us where we're currently using internal properties

### LINKED ISSUE #107: Remove PrintUses support
This was removed in newer R8 versions

### PR DESCRIPTION
**Pull request by ZacSweers**
- With Gradle 7.5 we can finally use new ArtifactCollection APIs that can be wired in as task inputs. This allows us to remove the previous hack of using an internal `configuration` property.
- Remove support for `PrintUses`, now everything runs through `TraceReferences` (the former was removed in R8)
- Move to version catalogs
- Fix implementation dependencies on kgp/kotlin-dsl
- Test with config cache enabled

Resolves #96
Resolves #107

Edit: This took much longer than expected because inexplicably, gradle 7.5 didn't pass on CI but 7.6 does ¯\_(ツ)_/¯.

### CODE REVIEW

**Comment by ZacSweers** [comment]
Hmmmm the proguardFiles view doesn't seem to work

File: keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt [hunk 1]
```diff
@@ -91,9 +85,9 @@ public class KeeperPlugin : Plugin<Project> {
   internal companion object {
     const val INTERMEDIATES_DIR = "intermediates/keeper"
     const val PRINTUSES_DEFAULT_VERSION = "1.6.53"
-    const val TRACE_REFERENCES_DEFAULT_VERSION = "3.0.9-dev"
+    const val TRACE_REFERENCES_DEFAULT_VERSION = "3.2.78"
     const val CONFIGURATION_NAME = "keeperR8"
-    private val MIN_GRADLE_VERSION = GradleVersion.version("7.2")
+    private val MIN_GRADLE_VERSION = GradleVersion.version("7.5")
```
Comment (autonomousapps) [inline keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt:90]: 7.5 or 7.6?

File: keeper-gradle-plugin/src/main/java/com/slack/keeper/VariantClasspathJar.kt [hunk 2]
```diff
@@ -151,7 +141,8 @@ public abstract class AndroidTestVariantClasspathJar : BaseKeeperJarTask() {
     val LOG = AndroidTestVariantClasspathJar::class.simpleName!!
   }
 
-  @get:PathSensitive(NONE) // Only care about the contents
+  @get:PathSensitive(NONE)
+  // Only care about the contents
```
Comment (autonomousapps) [inline keeper-gradle-plugin/src/main/java/com/slack/keeper/VariantClasspathJar.kt:145]: This comment looks like it should be up one line.

**Review by autonomousapps — APPROVED** [review]
LGTM.

File: keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt (hunk 1 above)
Comment (ZacSweers) [reply keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt:90]: 7.5, the new API was added in that

**Pull request merged**